circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/relations/relation_types.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
```

Comment:
I have mostly tried to make this more legible by giving things more accurate names. I think a deeper rework would be nice but there's not time now.